### ISSUE TITLE
Document that reassigning component references doesn’t work

### ISSUE THREAD

**Issue by gaearon** [gaearon/react-hot-loader#295]
As noted by @dferber90 in https://github.com/gaearon/react-hot-boilerplate/pull/61#issuecomment-218982382, there is a limitation even in RHL 3:

```js
let Stuff = () => ...
Stuff = connect()(Stuff)
export default Stuff
```

won’t be hot reloaded.

The fix is to make sure you don’t overwrite the original reference. For example:

```js
const Stuff = () => ...
export default connect()(Stuff)
```

or

```js
const Stuff = () => ...
const ConnectedStuff = connect()(Stuff)
export default ConnectedStuff
```

I don’t think it’s a big issue, but it’s a gotcha worth documenting.
It might become more of a problem when you chain HOCs but maybe we can live with that.

**Comment by josectello** [comment]
@gaearon, is this update to the docs still up for grabs, and if so, would it be something implemented into the `README.md`?

**Comment by calesce** [comment]
@josectello If you still want to do this, I'd say the best place is on the `next-docs` branch, in [`docs/README`](https://github.com/gaearon/react-hot-loader/blob/next-docs/docs/README.md).

There's also `troubleshooting.md`, but that'll need to be cleaned up, since most of it only applies to v1.

**Comment by calesce** [comment]
Closed by #402.
